CMakeLists.txt

```diff
@@ -73,7 +75,8 @@ option(BUILD_SHARED_LIBS "build shared libraries" ON)
 option(ENABLE_TESTING "build libdispatch tests" ON)
 
 if(CMAKE_SYSTEM_NAME STREQUAL Linux OR
-   CMAKE_SYSTEM_NAME STREQUAL Android)
+   CMAKE_SYSTEM_NAME STREQUAL Android OR
+   CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
```

Comment:
This would be a less intrusive patch if you put the conditional above Android rather than below. If you had it above then you'd only generate one line of addition:

```
 if(CMAKE_SYSTEM STREQUAL Linux OR
+   CMAKE_SYSTEM STREQUAL FreeBSD OR
   CMAKE_SYSTEM STREQUAL Android)
```